airdrop-machine/sources/airdrop_machine.move

```diff
@@ -0,0 +1,289 @@
+        let collection = object::object_from_constructor_ref(&constructor_ref);
+        configure_collection_and_token_properties(
+            &object_signer,
+            collection,
+            mutable_collection_metadata,
+            mutable_token_metadata,
+            tokens_burnable_by_collection_owner,
+            tokens_transferrable_by_collection_owner,
+        );
+
+        let object_signer = object::generate_signer(&constructor_ref);
+
+        move_to(&object_signer, CollectionConfig {
+            extend_ref: object::generate_extend_ref(object_constructor_ref),
+            ready_to_mint: false,
+        });
+
+        collection
+    }
+
+    fun royalty(
+        royalty_numerator: &mut Option<u64>, 
+        royalty_denominator: &mut Option<u64>, 
+        admin_addr: address
+    ): Option<Royalty> {
+        assert!(option::is_some(royalty_numerator) == option::is_some(royalty_denominator), error::invalid_argument(EROYALTY_CONFIGURATION_INVALID));
+        if (option::is_some(royalty_numerator) && option::is_some(royalty_denominator) && option::extract(royalty_numerator) != 0 && option::extract(royalty_denominator) != 0) {
+            option::some(royalty::create(option::extract(royalty_numerator), option::extract(royalty_denominator), admin_addr))
+        } else {
+            option::none()
+        }
+    }
+
+    fun configure_collection_and_token_properties(
+        admin: &signer,
+        collection: Object<Collection>,
+        mutable_collection_metadata: bool,
+        mutable_token_metadata: bool,
+        tokens_burnable_by_collection_owner: bool,
+        tokens_transferrable_by_collection_owner: bool,
+    ) {
+        collection_properties::set_mutable_description(admin, collection, mutable_collection_metadata);
+        collection_properties::set_mutable_uri(admin, collection, mutable_collection_metadata);
+        collection_properties::set_mutable_royalty(admin, collection, mutable_collection_metadata);
+        collection_properties::set_mutable_token_name(admin, collection, mutable_token_metadata);
+        collection_properties::set_mutable_token_properties(admin, collection, mutable_token_metadata);
+        collection_properties::set_mutable_token_description(admin, collection, mutable_token_metadata);
+        collection_properties::set_mutable_token_uri(admin, collection, mutable_token_metadata);
+        collection_properties::set_tokens_transferable_by_collection_owner(admin, collection, tokens_transferrable_by_collection_owner);
+        collection_properties::set_tokens_burnable_by_collection_owner(admin, collection, tokens_burnable_by_collection_owner);
+    }
+
+    fun assert_owner<T: key>(owner: address, object: Object<T>) {
+        assert!(object::owner(object) == owner || object::owns(object, owner), error::permission_denied(ENOT_OWNER));
+    }
+
+    fun collection_signer(collection: Object<Collection>): signer acquires CollectionConfig {
```

Comment:
Use the function from the token minter standard, and remove the refs here.

Reply:
I dont think we can as i pointed out in another thread(if im understanding signers & objs correctly), extend ref i need doesnt exist on collection signer but rather object signer that i created from creator.